packages/headless/src/features/case-assist/case-assist-analytics-actions.ts

```diff
@@ -31,128 +36,249 @@ export const logCaseNextStage = (
-  makeCaseAssistAnalyticsAction(
-    'analytics/caseAssist/classification/click',
-    (client, state) =>
-      client.logSelectFieldSuggestion({
+  makeCaseAssistAnalyticsAction({
+    prefix: 'analytics/caseAssist/classification/click',
+    __legacy__getBuilder: (client, state) => {
+      return client.logSelectFieldSuggestion({
         suggestion: caseAssistCaseClassificationSelector(
           state,
           classificationId
         ),
         ticket: caseAssistCaseSelector(state),
-      })
-  );
+      });
+    },
+    analyticsType: 'CaseAssist.SelectFieldClassification',
+    analyticsPayloadBuilder: (state): CaseAssist.SelectFieldClassification => {
+      return {
+        autoselected: false,
+        fieldClassification: {
+          id: classificationId,
+          responseId: state.caseField?.status?.lastResponseId || '',
+        },
+      };
+    },
+  });
 
 export const logDocumentSuggestionClick = (
   suggestionId: string
 ): CaseAssistAction =>
-  makeCaseAssistAnalyticsAction(
-    'analytics/caseAssist/documentSuggestion/click',
-    (client, state) =>
-      client.logSelectDocumentSuggestion({
+  makeCaseAssistAnalyticsAction({
+    prefix: 'analytics/caseAssist/documentSuggestion/click',
+    __legacy__getBuilder: (client, state) => {
+      return client.logSelectDocumentSuggestion({
         suggestion: caseAssistDocumentSuggestionSelector(state, suggestionId),
         ticket: caseAssistCaseSelector(state),
-      })
-  );
+      });
+    },
+    analyticsType: 'CaseAssist.DocumentSuggestionClick',
+    analyticsPayloadBuilder: (state): CaseAssist.DocumentSuggestionClick => {
+      return {
+        documentSuggestion: {
+          id: suggestionId,
+          responseId: state.documentSuggestion?.status?.lastResponseId || '',
+        },
+      };
+    },
+  });
 
 export const logQuickviewDocumentSuggestionClick = (
   suggestionId: string
 ): CaseAssistAction =>
-  makeCaseAssistAnalyticsAction(
-    'analytics/caseAssist/documentSuggestion/click',
-    (client, state) =>
-      client.logSelectDocumentSuggestion({
+  makeCaseAssistAnalyticsAction({
+    prefix: 'analytics/caseAssist/documentSuggestion/click',
+    __legacy__getBuilder: (client, state) => {
+      return client.logSelectDocumentSuggestion({
         suggestion: caseAssistDocumentSuggestionSelector(
           state,
           suggestionId,
           true
         ),
         ticket: caseAssistCaseSelector(state),
-      })
-  );
+      });
+    },
+    analyticsType: 'ItemClick',
+    analyticsPayloadBuilder: (state): ItemClick => {
+      const {
+        suggestion,
+        responseId,
+        suggestionId: uniqueId,
+      } = caseAssistDocumentSuggestionSelector(state, suggestionId, true);
+      return {
+        searchUid: responseId,
+        position: suggestion.documentPosition,
+        actionCause: 'preview',
+        itemMetadata: {
+          uniqueFieldName: 'uniqueId',
+          uniqueFieldValue: uniqueId,
```

Comment:
This is the uniqueId of the document suggestion? Is it just the permanentId? Or is there a separate doc suggestion Id returned by the API?

Reply:
Indeed this the uniqueId and this is the only doc suggestion id returned by the API.